src/h5web/dataset-visualizer/vis/MatrixVis.tsx

```diff
@@ -13,34 +21,179 @@ interface Props {
 
   const accessor: AccessorFunc =
     dims.length === 1 ? row => data[row] : (row, col) => data[row][col];
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function ValueCell(cellProps: CellProps): JSX.Element {
+    const { rowIndex: row, columnIndex: col, style } = cellProps;
+    return (
+      <div
+        className={styles.cell}
+        style={style}
+        data-bg={(col + row) % 2 === 1 ? '' : undefined}
+      >
+        {numeral(accessor(row, col)).format('0.000e+0')}
+      </div>
+    );
+  }
+
+  function IndexCell(indexCellProps: IndexCellProps): JSX.Element {
+    const { index, style } = indexCellProps;
+    return (
+      <div className={styles.indexCell} style={style}>
+        {index >= 0 ? (
+          index
+        ) : (
+          <button
+            className={styles.btn}
+            type="button"
+            onClick={() => {
+              setSticky(!sticky);
+            }}
+          >
+            <FiAnchor />
+          </button>
+        )}
+      </div>
+    );
+  }
+
+  function IndexColumn(): JSX.Element {
+    const { columnWidth, rowHeight, rowCount } = useContext(GridSettings);
+    const stickyStyle = sticky ? styles.sticky : '';
+    return (
+      <div
+        className={`${styles.indexColumn} ${stickyStyle}`}
+        style={{ width: columnWidth }}
+      >
+        {range(rowCount - 1).map(row => {
+          return (
+            <IndexCell
+              index={row}
+              key={`${row}.0`}
+              style={{
+                width: columnWidth,
+                height: rowHeight,
+              }}
+            />
+          );
+        })}
+      </div>
+    );
+  }
+
+  function IndexRow(): JSX.Element {
+    const { columnWidth, rowHeight, columnCount } = useContext(GridSettings);
+    const stickyStyle = sticky ? styles.sticky : '';
+    return (
+      <div
+        className={`${styles.indexRow} ${stickyStyle}`}
+        style={{ height: rowHeight }}
+      >
+        {range(columnCount).map(col => {
+          return (
+            <IndexCell
+              // Starts a -1 as the first column is the index column
+              index={col - 1}
+              key={`0.${col}`}
+              style={{
+                width: columnWidth,
+                height: rowHeight,
+              }}
+            />
+          );
+        })}
+      </div>
+    );
+  }
+
+  const innerElementType = forwardRef(({ children, ...rest }, ref) => {
+    return (
+      <div ref={ref as Ref<HTMLDivElement>} {...rest}>
+        <IndexRow />
+        <IndexColumn />
+        {children}
+      </div>
+    );
+  });
+
+  // IndexedGrid based on https://codesandbox.io/s/0mk3qwpl4l
+  function IndexedGrid(matrixProps: MatrixProps): JSX.Element {
+    const { children, width, height } = matrixProps;
+    const { columnWidth, rowHeight, columnCount, rowCount } = useContext(
+      GridSettings
+    );
+
+    return (
+      <Grid
+        itemData={{ NormalCellFunc: children }}
```

Comment:
If you inline the `CellRenderer` function here, directly inside `Grid` like this: `<Grid>{(...) => { ... }}</Grid>, perhaps you won't have to pass `NormalCellFunc` as `itemData` and then retrieve it in the cell renderer?